lib/trento/domain/cluster/cluster.ex

```diff
@@ -1,5 +1,51 @@
 defmodule Trento.Domain.Cluster do
-  @moduledoc false
+  @moduledoc """
+  The cluster aggregate manages all the domain logic related to
+  deployed HA Clusters (Pacemaker, Corosync, etc).
+  The HA cluster is used to handle the high availability scenarios on the installed
+  SAP infrastructure. That's why this domain is tailored to work on clusters managing
+  SAP workloads. Any other type of resource is not handled properly.
+
+  Each deployed cluster is registered as a new aggregate entry, meaning that all the hosts belonging
+  to the same cluster are part of the same stream. Among all the hosts, only the cluster discovery messages
+  coming from the **designated controller** update the aggregate. In any case, all the hosts are listed in
```

Comment:
> Among all the hosts, only the cluster discovery messages coming from the **designated controller** update the aggregate

Not really...

We take into account discoveries from non designated controllers to add them to the cluster, if it wasn't already registerd.
We do take into account those only after the designated controller was registered.

```elixir
# If no DC node was received yet, no cluster was registered.
def execute(%Cluster{cluster_id: nil}, %RegisterClusterHost{designated_controller: false}),
  do: {:error, :cluster_not_found}

def execute(
      %Cluster{} = cluster,
      %RegisterClusterHost{
        host_id: host_id,
        designated_controller: false
      }
    ) do
  maybe_emit_host_added_to_cluster_event(cluster, host_id)
end
```

Reply:
Ok, you are right. I meant that all the values "except" the host list are only updated in those conditions. That's why I added the next sentence. But I will make it more clear

Reply:
Rephrased